key.util/src/main/java/org/key_project/util/java/StringUtil.java

```diff
@@ -104,7 +110,7 @@ public static String trim(String text) {
      * @param text The text to convert.
      * @return The text in lower case or {@code null} if the given text is {@code null}.
      */
-    public static String toLowerCase(String text) {
+    public static @PolyNull String toLowerCase(@PolyNull String text) {
```

Comment:
See this discussion: https://github.com/KeYProject/key/pull/3183#pullrequestreview-1498952302